spring-mvc-java/src/main/java/com/baeldung/spring/web/controller/LoggerController.java

```diff
@@ -0,0 +1,37 @@
+package com.baeldung.spring.web.controller;
+
+import javax.servlet.ServletContext;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.web.context.support.WebApplicationContextUtils;
+
+import com.baeldung.spring.web.config.LogAccessService;
+import com.baeldung.spring.web.config.LogConfiguration;
+
+@Controller
+public class LoggerController {
+    @Autowired
+    private ServletContext servletContext;
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
```

Comment:
I tried to wire  LogAccessService logService but it failed with exception:

org.springframework.beans.factory.BeanCreationException: Error creating 
bean with name 'loggerController': Injection of autowired dependencies 
failed; nested exception is 
org.springframework.beans.factory.BeanCreationException: Could not 
autowire field: private com.baeldung.spring.web.config.LogAccessService 
com.baeldung.spring.web.controller.LoggerController.logService; nested 
exception is 
org.springframework.beans.factory.NoSuchBeanDefinitionException: No 
qualifying bean of type 
[com.baeldung.spring.web.config.LogAccessService] found for dependency: 
expected at least 1 bean which qualifies as autowire candidate for this 
dependency. Dependency annotations: 
{@org.springframework.beans.factory.annotation.Autowired(required=true)}

Hence I wired WebApplicationContext to get the bean from it.
I tried to understand why it did not find it, but could not work out.

ServletContext servletContext remained there to show in the article that 
it is possible to get root context with:

         final WebApplicationContext rootContext = 
WebApplicationContextUtils.getWebApplicationContext(servletContext);

If no need to describe how to get context we can remove this wiring.




Il 01/07/2016 15:32, slavisa-baeldung ha scritto:
>
> In 
> spring-mvc-java/src/main/java/com/baeldung/spring/web/controller/LoggerController.java 
> <https://github.com/eugenp/tutorials/pull/465#discussion_r69299613>:
>
> > +import org.springframework.web.bind.annotation.PathVariable;
> > +import org.springframework.web.bind.annotation.RequestMapping;
> > +import org.springframework.web.bind.annotation.ResponseBody;
> > +import org.springframework.web.context.WebApplicationContext;
> > +import org.springframework.web.context.support.WebApplicationContextUtils;
> > +
> > +import com.baeldung.spring.web.config.LogAccessService;
> > +import com.baeldung.spring.web.config.LogConfiguration;
> > +
> > +@Controller
> > +public class LoggerController {
> > +    @Autowired
> > +    private ServletContext servletContext;
> > +
> > +    @Autowired
> > +    private WebApplicationContext webApplicationContext;
>
> Don't get why are you wiring the context beans instead of service beans
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/eugenp/tutorials/pull/465/files/143861a2740c2145fecc4c005f7619f56d5f1709#r69299613>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AQK15nbReY4FtW5doBkTe3bUQnglB3K-ks5qRRcEgaJpZM4I_S2t>.
>